## Disease Symptom Prediction Dataset

In [1]:
import pandas as pd

In [2]:
dspd = pd.read_csv('Disease_Symptom_Prediction\dataset.csv')

In [3]:
dspd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4920 entries, 0 to 4919
Data columns (total 18 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Disease     4920 non-null   object
 1   Symptom_1   4920 non-null   object
 2   Symptom_2   4920 non-null   object
 3   Symptom_3   4920 non-null   object
 4   Symptom_4   4572 non-null   object
 5   Symptom_5   3714 non-null   object
 6   Symptom_6   2934 non-null   object
 7   Symptom_7   2268 non-null   object
 8   Symptom_8   1944 non-null   object
 9   Symptom_9   1692 non-null   object
 10  Symptom_10  1512 non-null   object
 11  Symptom_11  1194 non-null   object
 12  Symptom_12  744 non-null    object
 13  Symptom_13  504 non-null    object
 14  Symptom_14  306 non-null    object
 15  Symptom_15  240 non-null    object
 16  Symptom_16  192 non-null    object
 17  Symptom_17  72 non-null     object
dtypes: object(18)
memory usage: 692.0+ KB


In [4]:
dspd.describe()

Disease  Symptom_1  Symptom_2 Symptom_3    Symptom_4  \
count               4920       4920       4920      4920         4572   
unique                41         34         48        54           50   
top     Fungal infection   vomiting   vomiting   fatigue   high_fever   
freq                 120        822        870       726          378   

        Symptom_5 Symptom_6        Symptom_7        Symptom_8  \
count        3714      2934             2268             1944   
unique         38        32               26               21   
top      headache    nausea   abdominal_pain   abdominal_pain   
freq          348       390              264              276   

                 Symptom_9          Symptom_10     Symptom_11 Symptom_12  \
count                 1692                1512           1194        744   
unique                  22                  21             18         11   
top      yellowing_of_eyes   yellowing_of_eyes   irritability    malaise   
freq                   228                 198            120        126   

          Symptom_13   Symptom_14   Symptom_15        Symptom_16    Symptom_17  
count            504          306          240               192            72  
unique             8            4            3                 3             1  
top      muscle_pain   chest_pain   chest_pain   blood_in_sputum   muscle_pain  
freq              72           96          144                72            72

In [5]:
dspd.head()

Disease   Symptom_1              Symptom_2              Symptom_3  \
0  Fungal infection     itching              skin_rash   nodal_skin_eruptions   
1  Fungal infection   skin_rash   nodal_skin_eruptions    dischromic _patches   
2  Fungal infection     itching   nodal_skin_eruptions    dischromic _patches   
3  Fungal infection     itching              skin_rash    dischromic _patches   
4  Fungal infection     itching              skin_rash   nodal_skin_eruptions   

              Symptom_4 Symptom_5 Symptom_6 Symptom_7 Symptom_8 Symptom_9  \
0   dischromic _patches       NaN       NaN       NaN       NaN       NaN   
1                   NaN       NaN       NaN       NaN       NaN       NaN   
2                   NaN       NaN       NaN       NaN       NaN       NaN   
3                   NaN       NaN       NaN       NaN       NaN       NaN   
4                   NaN       NaN       NaN       NaN       NaN       NaN   

  Symptom_10 Symptom_11 Symptom_12 Symptom_13 Symptom_14 Symptom_15  \
0        NaN        NaN        NaN        NaN        NaN        NaN   
1        NaN        NaN        NaN        NaN        NaN        NaN   
2        NaN        NaN        NaN        NaN        NaN        NaN   
3        NaN        NaN        NaN        NaN        NaN        NaN   
4        NaN        NaN        NaN        NaN        NaN        NaN   

  Symptom_16 Symptom_17  
0        NaN        NaN  
1        NaN        NaN  
2        NaN        NaN  
3        NaN        NaN  
4        NaN        NaN

In [6]:
dspd.nunique()

Disease       41
Symptom_1     34
Symptom_2     48
Symptom_3     54
Symptom_4     50
Symptom_5     38
Symptom_6     32
Symptom_7     26
Symptom_8     21
Symptom_9     22
Symptom_10    21
Symptom_11    18
Symptom_12    11
Symptom_13     8
Symptom_14     4
Symptom_15     3
Symptom_16     3
Symptom_17     1
dtype: int64

Type of Diseases

In [7]:
dspd["Disease"].unique()

array(['Fungal infection', 'Allergy', 'GERD', 'Chronic cholestasis',
       'Drug Reaction', 'Peptic ulcer diseae', 'AIDS', 'Diabetes ',
       'Gastroenteritis', 'Bronchial Asthma', 'Hypertension ', 'Migraine',
       'Cervical spondylosis', 'Paralysis (brain hemorrhage)', 'Jaundice',
       'Malaria', 'Chicken pox', 'Dengue', 'Typhoid', 'hepatitis A',
       'Hepatitis B', 'Hepatitis C', 'Hepatitis D', 'Hepatitis E',
       'Alcoholic hepatitis', 'Tuberculosis', 'Common Cold', 'Pneumonia',
       'Dimorphic hemmorhoids(piles)', 'Heart attack', 'Varicose veins',
       'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia',
       'Osteoarthristis', 'Arthritis',
       '(vertigo) Paroymsal  Positional Vertigo', 'Acne',
       'Urinary tract infection', 'Psoriasis', 'Impetigo'], dtype=object)

In [8]:
dspd_disease = [i for i in dspd["Disease"].unique()]

In [9]:
print(dspd_disease)

['Fungal infection', 'Allergy', 'GERD', 'Chronic cholestasis', 'Drug Reaction', 'Peptic ulcer diseae', 'AIDS', 'Diabetes ', 'Gastroenteritis', 'Bronchial Asthma', 'Hypertension ', 'Migraine', 'Cervical spondylosis', 'Paralysis (brain hemorrhage)', 'Jaundice', 'Malaria', 'Chicken pox', 'Dengue', 'Typhoid', 'hepatitis A', 'Hepatitis B', 'Hepatitis C', 'Hepatitis D', 'Hepatitis E', 'Alcoholic hepatitis', 'Tuberculosis', 'Common Cold', 'Pneumonia', 'Dimorphic hemmorhoids(piles)', 'Heart attack', 'Varicose veins', 'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia', 'Osteoarthristis', 'Arthritis', '(vertigo) Paroymsal  Positional Vertigo', 'Acne', 'Urinary tract infection', 'Psoriasis', 'Impetigo']


Number of disease

In [10]:
no_disease =dspd["Disease"].nunique()
print("Number of disease: ", no_disease)

Number of disease:  41


### DSPD Disease value counts

In [11]:
dspd["Disease"].value_counts()

Disease
Fungal infection                           120
Hepatitis C                                120
Hepatitis E                                120
Alcoholic hepatitis                        120
Tuberculosis                               120
Common Cold                                120
Pneumonia                                  120
Dimorphic hemmorhoids(piles)               120
Heart attack                               120
Varicose veins                             120
Hypothyroidism                             120
Hyperthyroidism                            120
Hypoglycemia                               120
Osteoarthristis                            120
Arthritis                                  120
(vertigo) Paroymsal  Positional Vertigo    120
Acne                                       120
Urinary tract infection                    120
Psoriasis                                  120
Hepatitis D                                120
Hepatitis B                                120
Aller

In [12]:
dspd.shape

(4920, 18)

## Disease Symptoms and Patient Profile

In [13]:
dspp = pd.read_csv("Disease_Symptoms_and_Patient_Profile\Disease_symptom_and_patient_profile_dataset.csv")

In [14]:
dspp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 349 entries, 0 to 348
Data columns (total 10 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Disease               349 non-null    object
 1   Fever                 349 non-null    object
 2   Cough                 349 non-null    object
 3   Fatigue               349 non-null    object
 4   Difficulty Breathing  349 non-null    object
 5   Age                   349 non-null    int64 
 6   Gender                349 non-null    object
 7   Blood Pressure        349 non-null    object
 8   Cholesterol Level     349 non-null    object
 9   Outcome Variable      349 non-null    object
dtypes: int64(1), object(9)
memory usage: 27.4+ KB


In [15]:
dspp.describe()

Age
count  349.000000
mean    46.323782
std     13.085090
min     19.000000
25%     35.000000
50%     45.000000
75%     55.000000
max     90.000000

In [16]:
dspp.head()

Disease Fever Cough Fatigue Difficulty Breathing  Age  Gender  \
0    Influenza   Yes    No     Yes                  Yes   19  Female   
1  Common Cold    No   Yes     Yes                   No   25  Female   
2       Eczema    No   Yes     Yes                   No   25  Female   
3       Asthma   Yes   Yes      No                  Yes   25    Male   
4       Asthma   Yes   Yes      No                  Yes   25    Male   

  Blood Pressure Cholesterol Level Outcome Variable  
0            Low            Normal         Positive  
1         Normal            Normal         Negative  
2         Normal            Normal         Negative  
3         Normal            Normal         Positive  
4         Normal            Normal         Positive

In [17]:
dspp.nunique()

Disease                 116
Fever                     2
Cough                     2
Fatigue                   2
Difficulty Breathing      2
Age                      26
Gender                    2
Blood Pressure            3
Cholesterol Level         3
Outcome Variable          2
dtype: int64

Type of disease

In [18]:
dspp["Disease"].unique()

array(['Influenza', 'Common Cold', 'Eczema', 'Asthma', 'Hyperthyroidism',
       'Allergic Rhinitis', 'Anxiety Disorders', 'Diabetes',
       'Gastroenteritis', 'Pancreatitis', 'Rheumatoid Arthritis',
       'Depression', 'Liver Cancer', 'Stroke', 'Urinary Tract Infection',
       'Dengue Fever', 'Hepatitis', 'Kidney Cancer', 'Migraine',
       'Muscular Dystrophy', 'Sinusitis', 'Ulcerative Colitis',
       'Bipolar Disorder', 'Bronchitis', 'Cerebral Palsy',
       'Colorectal Cancer', 'Hypertensive Heart Disease',
       'Multiple Sclerosis', 'Myocardial Infarction (Heart...',
       'Urinary Tract Infection (UTI)', 'Osteoporosis', 'Pneumonia',
       'Atherosclerosis', 'Chronic Obstructive Pulmonary...', 'Epilepsy',
       'Hypertension', 'Obsessive-Compulsive Disorde...', 'Psoriasis',
       'Rubella', 'Cirrhosis', 'Conjunctivitis (Pink Eye)',
       'Liver Disease', 'Malaria', 'Spina Bifida', 'Kidney Disease',
       'Osteoarthritis', 'Klinefelter Syndrome', 'Acne', 'Brain Tumor',


In [19]:
dspp_disease = [i for i in dspp["Disease"].unique()]
print(dspp_disease)

['Influenza', 'Common Cold', 'Eczema', 'Asthma', 'Hyperthyroidism', 'Allergic Rhinitis', 'Anxiety Disorders', 'Diabetes', 'Gastroenteritis', 'Pancreatitis', 'Rheumatoid Arthritis', 'Depression', 'Liver Cancer', 'Stroke', 'Urinary Tract Infection', 'Dengue Fever', 'Hepatitis', 'Kidney Cancer', 'Migraine', 'Muscular Dystrophy', 'Sinusitis', 'Ulcerative Colitis', 'Bipolar Disorder', 'Bronchitis', 'Cerebral Palsy', 'Colorectal Cancer', 'Hypertensive Heart Disease', 'Multiple Sclerosis', 'Myocardial Infarction (Heart...', 'Urinary Tract Infection (UTI)', 'Osteoporosis', 'Pneumonia', 'Atherosclerosis', 'Chronic Obstructive Pulmonary...', 'Epilepsy', 'Hypertension', 'Obsessive-Compulsive Disorde...', 'Psoriasis', 'Rubella', 'Cirrhosis', 'Conjunctivitis (Pink Eye)', 'Liver Disease', 'Malaria', 'Spina Bifida', 'Kidney Disease', 'Osteoarthritis', 'Klinefelter Syndrome', 'Acne', 'Brain Tumor', 'Cystic Fibrosis', 'Glaucoma', 'Rabies', 'Chickenpox', 'Coronary Artery Disease', 'Eating Disorders (Ano

Number of disease

In [20]:
unique_disease = dspp["Disease"].nunique()
print("Number of diseases: ", unique_disease)

Number of diseases:  116


### DSPP Disease value counts

In [21]:
dspp["Disease"].value_counts()

Disease
Asthma                            23
Stroke                            16
Osteoporosis                      14
Hypertension                      10
Diabetes                          10
                                  ..
Autism Spectrum Disorder (ASD)     1
Hypoglycemia                       1
Fibromyalgia                       1
Eating Disorders (Anorexia,...     1
Williams Syndrome                  1
Name: count, Length: 116, dtype: int64

In [22]:
print([i for i in dspp["Disease"].value_counts()])

[23, 16, 14, 10, 10, 10, 8, 8, 8, 7, 7, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 4, 3, 3, 3, 3, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]


## Symptom Checker

In [23]:
sc_train = pd.read_csv("Symptom_checker\Training.csv")
sc_test = pd.read_csv("Symptom_checker\Testing.csv")

In [24]:
sc_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4920 entries, 0 to 4919
Columns: 133 entries, itching to prognosis
dtypes: int64(132), object(1)
memory usage: 5.0+ MB


In [25]:
sc_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Columns: 133 entries, itching to prognosis
dtypes: int64(132), object(1)
memory usage: 42.7+ KB


In [26]:
sc_train.describe()

itching    skin_rash  nodal_skin_eruptions  continuous_sneezing  \
count  4920.000000  4920.000000           4920.000000          4920.000000   
mean      0.137805     0.159756              0.021951             0.045122   
std       0.344730     0.366417              0.146539             0.207593   
min       0.000000     0.000000              0.000000             0.000000   
25%       0.000000     0.000000              0.000000             0.000000   
50%       0.000000     0.000000              0.000000             0.000000   
75%       0.000000     0.000000              0.000000             0.000000   
max       1.000000     1.000000              1.000000             1.000000   

         shivering       chills   joint_pain  stomach_pain      acidity  \
count  4920.000000  4920.000000  4920.000000   4920.000000  4920.000000   
mean      0.021951     0.162195     0.139024      0.045122     0.045122   
std       0.146539     0.368667     0.346007      0.207593     0.207593   
min       0.000000     0.000000     0.000000      0.000000     0.000000   
25%       0.000000     0.000000     0.000000      0.000000     0.000000   
50%       0.000000     0.000000     0.000000      0.000000     0.000000   
75%       0.000000     0.000000     0.000000      0.000000     0.000000   
max       1.000000     1.000000     1.000000      1.000000     1.000000   

       ulcers_on_tongue  ...  pus_filled_pimples   blackheads     scurring  \
count       4920.000000  ...         4920.000000  4920.000000  4920.000000   
mean           0.021951  ...            0.021951     0.021951     0.021951   
std            0.146539  ...            0.146539     0.146539     0.146539   
min            0.000000  ...            0.000000     0.000000     0.000000   
25%            0.000000  ...            0.000000     0.000000     0.000000   
50%            0.000000  ...            0.000000     0.000000     0.000000   
75%            0.000000  ...            0.000000     0.000000     0.000000   
max            1.000000  ...            1.000000     1.000000     1.000000   

       skin_peeling  silver_like_dusting  small_dents_in_nails  \
count   4920.000000          4920.000000           4920.000000   
mean       0.023171             0.023171              0.023171   
std        0.150461             0.150461              0.150461   
min        0.000000             0.000000              0.000000   
25%        0.000000             0.000000              0.000000   
50%        0.000000             0.000000              0.000000   
75%        0.000000             0.000000              0.000000   
max        1.000000             1.000000              1.000000   

       inflammatory_nails      blister  red_sore_around_nose  \
count         4920.000000  4920.000000           4920.000000   
mean             0.023171     0.023171              0.023171   
std              0.150461     0.150461              0.150461   
min              0.000000     0.000000              0.000000   
25%              0.000000     0.000000              0.000000   
50%              0.000000     0.000000              0.000000   
75%              0.000000     0.000000              0.000000   
max              1.000000     1.000000              1.000000   

       yellow_crust_ooze  
count        4920.000000  
mean            0.023171  
std             0.150461  
min             0.000000  
25%             0.000000  
50%             0.000000  
75%             0.000000  
max             1.000000  

[8 rows x 132 columns]

In [27]:
sc_test.describe()

itching  skin_rash  nodal_skin_eruptions  continuous_sneezing  \
count  41.000000  41.000000             41.000000            41.000000   
mean    0.146341   0.170732              0.024390             0.048780   
std     0.357839   0.380949              0.156174             0.218085   
min     0.000000   0.000000              0.000000             0.000000   
25%     0.000000   0.000000              0.000000             0.000000   
50%     0.000000   0.000000              0.000000             0.000000   
75%     0.000000   0.000000              0.000000             0.000000   
max     1.000000   1.000000              1.000000             1.000000   

       shivering     chills  joint_pain  stomach_pain    acidity  \
count  41.000000  41.000000   41.000000     41.000000  41.000000   
mean    0.024390   0.170732    0.146341      0.048780   0.048780   
std     0.156174   0.380949    0.357839      0.218085   0.218085   
min     0.000000   0.000000    0.000000      0.000000   0.000000   
25%     0.000000   0.000000    0.000000      0.000000   0.000000   
50%     0.000000   0.000000    0.000000      0.000000   0.000000   
75%     0.000000   0.000000    0.000000      0.000000   0.000000   
max     1.000000   1.000000    1.000000      1.000000   1.000000   

       ulcers_on_tongue  ...  pus_filled_pimples  blackheads   scurring  \
count         41.000000  ...           41.000000   41.000000  41.000000   
mean           0.024390  ...            0.024390    0.024390   0.024390   
std            0.156174  ...            0.156174    0.156174   0.156174   
min            0.000000  ...            0.000000    0.000000   0.000000   
25%            0.000000  ...            0.000000    0.000000   0.000000   
50%            0.000000  ...            0.000000    0.000000   0.000000   
75%            0.000000  ...            0.000000    0.000000   0.000000   
max            1.000000  ...            1.000000    1.000000   1.000000   

       skin_peeling  silver_like_dusting  small_dents_in_nails  \
count     41.000000            41.000000             41.000000   
mean       0.024390             0.024390              0.024390   
std        0.156174             0.156174              0.156174   
min        0.000000             0.000000              0.000000   
25%        0.000000             0.000000              0.000000   
50%        0.000000             0.000000              0.000000   
75%        0.000000             0.000000              0.000000   
max        1.000000             1.000000              1.000000   

       inflammatory_nails    blister  red_sore_around_nose  yellow_crust_ooze  
count           41.000000  41.000000             41.000000          41.000000  
mean             0.024390   0.024390              0.024390           0.024390  
std              0.156174   0.156174              0.156174           0.156174  
min              0.000000   0.000000              0.000000           0.000000  
25%              0.000000   0.000000              0.000000           0.000000  
50%              0.000000   0.000000              0.000000           0.000000  
75%              0.000000   0.000000              0.000000           0.000000  
max              1.000000   1.000000              1.000000           1.000000  

[8 rows x 132 columns]

In [28]:
sc_train.head()

itching  skin_rash  nodal_skin_eruptions  continuous_sneezing  shivering  \
0        1          1                     1                    0          0   
1        0          1                     1                    0          0   
2        1          0                     1                    0          0   
3        1          1                     0                    0          0   
4        1          1                     1                    0          0   

   chills  joint_pain  stomach_pain  acidity  ulcers_on_tongue  ...  \
0       0           0             0        0                 0  ...   
1       0           0             0        0                 0  ...   
2       0           0             0        0                 0  ...   
3       0           0             0        0                 0  ...   
4       0           0             0        0                 0  ...   

   blackheads  scurring  skin_peeling  silver_like_dusting  \
0           0         0             0                    0   
1           0         0             0                    0   
2           0         0             0                    0   
3           0         0             0                    0   
4           0         0             0                    0   

   small_dents_in_nails  inflammatory_nails  blister  red_sore_around_nose  \
0                     0                   0        0                     0   
1                     0                   0        0                     0   
2                     0                   0        0                     0   
3                     0                   0        0                     0   
4                     0                   0        0                     0   

   yellow_crust_ooze         prognosis  
0                  0  Fungal infection  
1                  0  Fungal infection  
2                  0  Fungal infection  
3                  0  Fungal infection  
4                  0  Fungal infection  

[5 rows x 133 columns]

In [29]:
sc_test.head()

itching  skin_rash  nodal_skin_eruptions  continuous_sneezing  shivering  \
0        1          1                     1                    0          0   
1        0          0                     0                    1          1   
2        0          0                     0                    0          0   
3        1          0                     0                    0          0   
4        1          1                     0                    0          0   

   chills  joint_pain  stomach_pain  acidity  ulcers_on_tongue  ...  \
0       0           0             0        0                 0  ...   
1       1           0             0        0                 0  ...   
2       0           0             1        1                 1  ...   
3       0           0             0        0                 0  ...   
4       0           0             1        0                 0  ...   

   blackheads  scurring  skin_peeling  silver_like_dusting  \
0           0         0             0                    0   
1           0         0             0                    0   
2           0         0             0                    0   
3           0         0             0                    0   
4           0         0             0                    0   

   small_dents_in_nails  inflammatory_nails  blister  red_sore_around_nose  \
0                     0                   0        0                     0   
1                     0                   0        0                     0   
2                     0                   0        0                     0   
3                     0                   0        0                     0   
4                     0                   0        0                     0   

   yellow_crust_ooze            prognosis  
0                  0     Fungal infection  
1                  0              Allergy  
2                  0                 GERD  
3                  0  Chronic cholestasis  
4                  0        Drug Reaction  

[5 rows x 133 columns]

Training - type of diseases

In [30]:
sc_train["prognosis"].unique()

array(['Fungal infection', 'Allergy', 'GERD', 'Chronic cholestasis',
       'Drug Reaction', 'Peptic ulcer diseae', 'AIDS', 'Diabetes ',
       'Gastroenteritis', 'Bronchial Asthma', 'Hypertension ', 'Migraine',
       'Cervical spondylosis', 'Paralysis (brain hemorrhage)', 'Jaundice',
       'Malaria', 'Chicken pox', 'Dengue', 'Typhoid', 'hepatitis A',
       'Hepatitis B', 'Hepatitis C', 'Hepatitis D', 'Hepatitis E',
       'Alcoholic hepatitis', 'Tuberculosis', 'Common Cold', 'Pneumonia',
       'Dimorphic hemmorhoids(piles)', 'Heart attack', 'Varicose veins',
       'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia',
       'Osteoarthristis', 'Arthritis',
       '(vertigo) Paroymsal  Positional Vertigo', 'Acne',
       'Urinary tract infection', 'Psoriasis', 'Impetigo'], dtype=object)

Testing - type of diseases

In [31]:
sc_test["prognosis"].unique()

array(['Fungal infection', 'Allergy', 'GERD', 'Chronic cholestasis',
       'Drug Reaction', 'Peptic ulcer diseae', 'AIDS', 'Diabetes ',
       'Gastroenteritis', 'Bronchial Asthma', 'Hypertension ', 'Migraine',
       'Cervical spondylosis', 'Paralysis (brain hemorrhage)', 'Jaundice',
       'Malaria', 'Chicken pox', 'Dengue', 'Typhoid', 'hepatitis A',
       'Hepatitis B', 'Hepatitis C', 'Hepatitis D', 'Hepatitis E',
       'Alcoholic hepatitis', 'Tuberculosis', 'Common Cold', 'Pneumonia',
       'Dimorphic hemmorhoids(piles)', 'Heart attack', 'Varicose veins',
       'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia',
       'Osteoarthristis', 'Arthritis',
       '(vertigo) Paroymsal  Positional Vertigo', 'Acne',
       'Urinary tract infection', 'Psoriasis', 'Impetigo'], dtype=object)

In [32]:
sc_train_disease = [i for i in sc_train["prognosis"].unique()]
sc_test_disease = [i for i in sc_test["prognosis"].unique()]

Training - number of diseases

In [33]:
sc_train_numb = sc_train["prognosis"].nunique()
print("Number of diseases in training dataset: ", sc_train_numb)

Number of diseases in training dataset:  41


### SC_Train disease value counts

In [34]:
sc_train["prognosis"].value_counts()

prognosis
Fungal infection                           120
Hepatitis C                                120
Hepatitis E                                120
Alcoholic hepatitis                        120
Tuberculosis                               120
Common Cold                                120
Pneumonia                                  120
Dimorphic hemmorhoids(piles)               120
Heart attack                               120
Varicose veins                             120
Hypothyroidism                             120
Hyperthyroidism                            120
Hypoglycemia                               120
Osteoarthristis                            120
Arthritis                                  120
(vertigo) Paroymsal  Positional Vertigo    120
Acne                                       120
Urinary tract infection                    120
Psoriasis                                  120
Hepatitis D                                120
Hepatitis B                                120
All

Testing - number of diseases

In [35]:
sc_test_numb = sc_test["prognosis"].nunique()
print("Number of diseases in testing dataset: ", sc_test_numb)

Number of diseases in testing dataset:  41


### SC_Train disease value counts

In [36]:
sc_test["prognosis"].value_counts()

prognosis
Fungal infection                           1
Hepatitis C                                1
Hepatitis E                                1
Alcoholic hepatitis                        1
Tuberculosis                               1
Common Cold                                1
Pneumonia                                  1
Dimorphic hemmorhoids(piles)               1
Heart attack                               1
Varicose veins                             1
Hypothyroidism                             1
Hyperthyroidism                            1
Hypoglycemia                               1
Osteoarthristis                            1
Arthritis                                  1
(vertigo) Paroymsal  Positional Vertigo    1
Acne                                       1
Urinary tract infection                    1
Psoriasis                                  1
Hepatitis D                                1
Hepatitis B                                1
Allergy                                    1


### Compare disease similarity

In [37]:
list_same = []

for i in range(len(dspd_disease)):
    for j in range(len(sc_train_disease)):
        if dspd_disease[i] == sc_train_disease[j]:
            list_same.append(dspd_disease[i])


In [38]:
print(list_same)
print(len(list_same))

['Fungal infection', 'Allergy', 'GERD', 'Chronic cholestasis', 'Drug Reaction', 'Peptic ulcer diseae', 'AIDS', 'Diabetes ', 'Gastroenteritis', 'Bronchial Asthma', 'Hypertension ', 'Migraine', 'Cervical spondylosis', 'Paralysis (brain hemorrhage)', 'Jaundice', 'Malaria', 'Chicken pox', 'Dengue', 'Typhoid', 'hepatitis A', 'Hepatitis B', 'Hepatitis C', 'Hepatitis D', 'Hepatitis E', 'Alcoholic hepatitis', 'Tuberculosis', 'Common Cold', 'Pneumonia', 'Dimorphic hemmorhoids(piles)', 'Heart attack', 'Varicose veins', 'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia', 'Osteoarthristis', 'Arthritis', '(vertigo) Paroymsal  Positional Vertigo', 'Acne', 'Urinary tract infection', 'Psoriasis', 'Impetigo']
41


In [39]:
list_same_2 = []

for i in range(len(dspd_disease)):
    for j in range(len(dspp_disease)):
        if dspd_disease[i] == dspp_disease[j]:
            list_same_2.append(dspd_disease[i])

In [40]:
print(list_same_2)
print(len(list_same_2))

['Gastroenteritis', 'Migraine', 'Malaria', 'Hepatitis B', 'Tuberculosis', 'Common Cold', 'Pneumonia', 'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia', 'Acne', 'Psoriasis']
12


## Summary

### Disease Symptom Prediction Database (DSPD)
41 diseases, 120 rows for each disease
Same as SC

### Disease Symptom & Patient Profile
116 diseases, range from 23, 16, 14, 10, 8, 7, 6, 5, 4, 3, 2, 1

### Symptom Checker(SC)
41 diseases, 120 r0ws for each disease
Same as DSPD
Split into train and test datasets

### DSPD Diseases

In [41]:
dspdDisease = [i for i in dspd["Disease"].unique()]

In [42]:
for i in range(len(dspdDisease)):
    print(i+1, ": ", dspdDisease[i])

1 :  Fungal infection
2 :  Allergy
3 :  GERD
4 :  Chronic cholestasis
5 :  Drug Reaction
6 :  Peptic ulcer diseae
7 :  AIDS
8 :  Diabetes 
9 :  Gastroenteritis
10 :  Bronchial Asthma
11 :  Hypertension 
12 :  Migraine
13 :  Cervical spondylosis
14 :  Paralysis (brain hemorrhage)
15 :  Jaundice
16 :  Malaria
17 :  Chicken pox
18 :  Dengue
19 :  Typhoid
20 :  hepatitis A
21 :  Hepatitis B
22 :  Hepatitis C
23 :  Hepatitis D
24 :  Hepatitis E
25 :  Alcoholic hepatitis
26 :  Tuberculosis
27 :  Common Cold
28 :  Pneumonia
29 :  Dimorphic hemmorhoids(piles)
30 :  Heart attack
31 :  Varicose veins
32 :  Hypothyroidism
33 :  Hyperthyroidism
34 :  Hypoglycemia
35 :  Osteoarthristis
36 :  Arthritis
37 :  (vertigo) Paroymsal  Positional Vertigo
38 :  Acne
39 :  Urinary tract infection
40 :  Psoriasis
41 :  Impetigo


### Unique Symptoms for DSPD

In [43]:
unique_symptoms = set()
for i, column in enumerate(dspd.columns):
  unique_symptoms |= set(dspd[column])

print(unique_symptoms)
print(len(unique_symptoms))

{nan, ' loss_of_appetite', ' spinning_movements', ' fatigue', 'GERD', ' stomach_pain', ' weight_loss', ' dischromic _patches', ' swollen_extremeties', ' bruising', ' blurred_and_distorted_vision', ' yellowing_of_eyes', ' drying_and_tingling_lips', ' nodal_skin_eruptions', ' spotting_ urination', ' passage_of_gases', ' abnormal_menstruation', ' blister', ' mucoid_sputum', ' acidity', ' rusty_sputum', ' vomiting', ' toxic_look_(typhos)', ' enlarged_thyroid', ' redness_of_eyes', ' phlegm', ' watering_from_eyes', 'Osteoarthristis', ' weakness_of_one_body_side', ' muscle_pain', ' weakness_in_limbs', ' cough', ' knee_pain', ' pain_behind_the_eyes', '(vertigo) Paroymsal  Positional Vertigo', 'Hyperthyroidism', ' skin_peeling', 'itching', ' joint_pain', ' bladder_discomfort', ' small_dents_in_nails', ' extra_marital_contacts', ' inflammatory_nails', ' cramps', ' swollen_legs', 'Drug Reaction', ' swelling_of_stomach', ' distention_of_abdomen', ' increased_appetite', ' slurred_speech', ' altered

In [44]:
len(unique_symptoms)

173

In [ ]:
df_symptoms = dspd.groupby('Disease').agg(lambda x: set(x.dropna()))
df_symptoms

In [53]:
symptoms_col = dspd.iloc[:, 1:]
fungal = set(dspd.loc[dspd['Disease'] == 'Fungal Infection', dspd.columns[1:]].values.flatten())
print(fungal)

set()
